packages/remix/src/utils.ts

```diff
@@ -273,3 +276,96 @@ export function addDependency(
   }
   return spawnAsync(cliType, args.concat(names), opts);
 }
+
+export async function ensureResolvable(
+  start: string,
+  base: string,
+  pkgName: string
+): Promise<string> {
+  try {
+    const resolvedPkgPath = _require.resolve(`${pkgName}/package.json`, {
+      paths: [start],
+    });
+    const resolvedPath = dirname(resolvedPkgPath);
+    if (!relative(base, resolvedPath).startsWith(`..${sep}`)) {
+      // Resolved path is within the root of the project, so all good
+      debug(`"${pkgName}" resolved to '${resolvedPath}'`);
+      return resolvedPath;
+    }
+  } catch (err: any) {
+    if (err.code !== 'MODULE_NOT_FOUND') {
+      throw err;
+    }
+  }
+
+  // If we got to here then `pkgName` was not resolvable up to the root
+  // of the project. Try a couple symlink tricks, otherwise we'll bail.
+
+  // Attempt to find the package in `node_modules/.pnpm` (pnpm)
+  const pnpmDir = await walkParentDirs({
+    base,
+    start,
+    filename: 'node_modules/.pnpm',
+  });
+  if (pnpmDir) {
+    const prefix = `${pkgName.replace('/', '+')}@`;
+    const packages = await fs.readdir(pnpmDir);
+    const match = packages.find(p => p.startsWith(prefix));
+    if (match) {
+      const pkgDir = join(pnpmDir, match, 'node_modules', pkgName);
+      await ensureSymlink(pkgDir, join(start, 'node_modules'), pkgName);
```

Comment:
```diff
-await ensureSymlink(pkgDir, join(pnpmDir, '..'), pkgName);
+await ensureSymlink(pkgDir, join(start, 'node_modules'), pkgName);
```

This is the main fix in this PR. The rest of the diff was moving the function to the `utils` directory and adding a unit test for the updated behavior.